client/rest-high-level/src/main/java/org/elasticsearch/client/asyncsearch/SubmitAsyncSearchRequest.java

```diff
@@ -0,0 +1,211 @@
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * A request to track asynchronously the progress of a search against one or more indices.
+ */
+public class SubmitAsyncSearchRequest implements Validatable {
+
+    public static long MIN_KEEP_ALIVE = TimeValue.timeValueMinutes(1).millis();
+
+    private TimeValue waitForCompletion;
+    private Boolean cleanOnCompletion;
+    private TimeValue keepAlive;
+    private final SearchRequest searchRequest;
+
+    /**
+     * Creates a new request
+     */
+    public SubmitAsyncSearchRequest(SearchSourceBuilder source, String... indices) {
+        this.searchRequest = new SearchRequest(indices, source);
+        searchRequest.setCcsMinimizeRoundtrips(false);
+        searchRequest.setPreFilterShardSize(1);
+        searchRequest.setBatchedReduceSize(5);
+        searchRequest.requestCache(true);
+    }
+
+
+    /**
+     * Get the {@link SearchRequest} that this submit request wraps
+     */
+    public SearchRequest getSearchRequest() {
+        return searchRequest;
+    }
+
+    /**
+     * Get the minimum time that the request should wait before returning a partial result (defaults to 1 second).
+     */
+    public TimeValue getWaitForCompletion() {
+        return waitForCompletion;
+    }
+
+    /**
+     * Sets the minimum time that the request should wait before returning a partial result (defaults to 1 second).
+     */
+    public void setWaitForCompletion(TimeValue waitForCompletion) {
+        this.waitForCompletion = waitForCompletion;
+    }
+
+    /**
+     * Returns whether the resource resource should be removed on completion or failure (defaults to true).
+     */
+    public Boolean isCleanOnCompletion() {
+        return cleanOnCompletion;
+    }
+
+    /**
+     * Determines if the resource should be removed on completion or failure (defaults to true).
+     */
+    public void setCleanOnCompletion(boolean cleanOnCompletion) {
+        this.cleanOnCompletion = cleanOnCompletion;
+    }
+
+    /**
+     * Get the amount of time after which the result will expire (defaults to 5 days).
+     */
+    public TimeValue getKeepAlive() {
+        return keepAlive;
+    }
+
+    /**
+     * Sets the amount of time after which the result will expire (defaults to 5 days).
+     */
+    public void setKeepAlive(TimeValue keepAlive) {
+        this.keepAlive = keepAlive;
+    }
+
+    // setters for request parameters of the wrapped SearchRequest
+    /**
+     * Set the routing values to control the shards that the search will be executed on.
+     * A comma separated list of routing values to control the shards the search will be executed on.
+     */
+    public void routing(String routing) {
+        this.searchRequest.routing(routing);
+    }
+
+    /**
+     * Set the routing values to control the shards that the search will be executed on.
+     */
+    public void routing(String... routings) {
```

Comment:
annoying nit: I see that some getters and setters use the get and set prefix in this class, while others don't. I get why, but shall we unify the style at least throughout this one class?